### ISSUE TITLE
Develop Time Tracker

### ISSUE THREAD

**Issue by sarL3y** [hackforla/VRMS#192]
### Overview
Develop and implement a time tracker to provide users a way to gauge their time with the organization

### Action Items

**Comment by switchzer0** [comment]
functionality recommendation: think it would be cool for people to get asked 1x/week when they check in to meetings to estimate how much solo time they spent on the org. could validate by highlighting people who estimate 1 or 2 SD above/below mean by project/brigade/etc. to project lead to follow up for accountability

**Comment by ExperimentsInHonesty** [comment]
@patrickgus wire this up to personal dashboard /profile

**Comment by patrickgus** [comment]
1. Progress: Getting started
2. Blockers: Need to work with @rachelwhaley to create data model for activity logs
3. Availability: Weekdays after 3pm
4. ETA: n/a

**Comment by patrickgus** [comment]
1. Progress: Did work on the front end. Added section to profile page where activity log will be rendered.
2. Blockers: Need to get activity log data model from Rachel @rachelwhaley to hook up back end.
3. Availability: Weekdays after 3pm
4. ETA: 2 weeks

**Comment by alex-anakin** [comment]
@patrickgus please take a look at https://github.com/hackforla/VRMS/pull/239

**Comment by patrickgus** [comment]
1. Progress: Did some more work on the front end.
2. Blockers: Waiting on pull request #239 to be merged to hook up back end.
3. Availability: Weekdays after 3pm
4. ETA: 2 weeks

**Comment by patrickgus** [comment]
1. Progress: None
2. Blockers: #244 
3. Availability: Weekdays after 3pm
4. ETA: N/A

**Comment by patrickgus** [comment]
1. Progress: None
2. Blockers: #244
3. Availability: Weekdays after 3pm
4. ETA: N/A

**Comment by patrickgus** [comment]
1. Progress: None
2. Blockers: #244
3. Availability: Weekdays after 3pm
4. ETA: N/A

**Comment by patrickgus** [comment]
1. Progress: None
2. Blockers: Waiting on PR #239 
3. Availability: Weekdays after 3pm
4. ETA: N/A

**Comment by patrickgus** [comment]
1. Progress: Created GET and POST routes for the backend. Also created test for the data model.
2. Blockers: Need to get test data in db in order to test the routes. Not sure how to create a collection in Mongo or which db to use.
3. Availability: Weekdays after 3pm
4. ETA: 1-2 weeks

**Comment by patrickgus** [comment]
Progress: Created test data collection in database but having trouble retrieving data using GET requests. 
Blockers: None
Availability: Weekdays after 3pm
ETA: 1-2 weeks

**Comment by patrickgus** [comment]
Progress: None. Still need help with accessing data from test db
Blockers: None
Availability: Weekdays after 3pm
ETA: 1-2 weeks

**Comment by patrickgus** [comment]
Progress: Able to access data from db now using routes. Working on getting the front end hooked up to backend. Having trouble getting logs to render.
Blockers: None
Availability: Weekdays after 3pm
ETA: 1-2 weeks
Notes: How should we go about calculating hours? (moment?) Will it be a problem for admins to access the total hours logged by users if it is calculated in the front end?

**Comment by patrickgus** [comment]
Progress: Was able to get a few test activity logs to render on the user profile. Next is to create a from for adding logs. 
Blockers: None
Availability: Weekdays after 3pm
ETA: 1 week
Questions: Do we want a delete and/or edit functionality for each activity log?

**Comment by patrickgus** [comment]
Progress: Started to create a form to add new logs
Blockers: None
Availability: Weekdays after 3pm
ETA: 1 week

**Comment by patrickgus** [comment]
Progress: Finished the form for adding new logs and added a link to the form on the user profile page.
Blockers: None
Availability: Weekdays after 3pm
ETA: 1 week

**Comment by patrickgus** [comment]
Progress: Working on hooking up the back end to the front end form so that the newly added logs will render on the profile page.
Blockers: None
Availability: Weekdays after 3pm
ETA: 1 week

**Comment by patrickgus** [comment]
Progress: Refactoring so that it is contained in its own component. Still need to hook up to the back end.
Blockers: None
Availability: Weekdays after 3pm
ETA: 1 week

**Comment by patrickgus** [comment]
Progress: Done refactoring form for adding new logs. Working on getting newly added log to render on the profile page.
Blockers: None
Availability: Very limited this week
ETA: 2 weeks

**Comment by patrickgus** [comment]
Progress: None
Blockers: Running into a proxy error on my local machine where I can't load anything from the backend. Not sure what is causing error but it occurred after I pulled changes from the development branch.
Availability: Only Tuesday after 3pm this week.

**Comment by switchzer0** [comment]
haven't heard from patrick for several months; I believe this work is obsolete as not on the critical path for v0.4

**Comment by peteplass** [comment]
@switchzer0 Could you please elaborate on why this is now in Review and assign someone to review? Is Patrick still in on this?

**Comment by peteplass** [comment]
Closing because no longer relevant.
